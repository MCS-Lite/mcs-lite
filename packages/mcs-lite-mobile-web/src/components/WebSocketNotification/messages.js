import { defineMessages } from 'react-intl';

const messages = defineMessages({
  'WebSocketNotification.info': {
    id: 'WebSocketNotification.info',
    description: 'Error message',
    defaultMessage: '伺服器連線異常，目前無法即時上傳與接收資料，是否要重新連線？',
  },
  'WebSocketNotification.reconnect': {
    id: 'WebSocketNotification.reconnect',
    description: 'For button content',
    defaultMessage: '重新連線',
  },
});

export default messages;
