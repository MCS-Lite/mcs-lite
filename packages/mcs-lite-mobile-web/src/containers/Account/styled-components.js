import styled from 'styled-components';
import Button from 'mcs-lite-ui/lib/Button';
import P from 'mcs-lite-ui/lib/P';
import Logo from '../../components/Logo';

export const FlatButton = styled(Button)`
  border: initial;
  border-radius: initial;
  height: 56px;
  border-top: 1px solid ${props => props.theme.color.white};
`;

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  flex-grow: 1;
`;

export const Body = styled.div`
  width: 100%;
  flex-grow: 1;
  display: flex;
  align-items: center;
  flex-direction: column;

  > *:not(:first-child) {
    margin-bottom: 8px;
  }
`;

export const StyledLogo = styled(Logo)`
  margin-bottom: 32px;
`;

export const VersionWrapper = styled(P)`
  text-align: center;
  margin-bottom: 8px;
`;

export const Footer = styled.footer`
  width: 100%;
`;
