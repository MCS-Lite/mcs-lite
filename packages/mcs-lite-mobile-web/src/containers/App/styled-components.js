import styled from 'styled-components';
import { MobileContentWrapper } from 'mcs-lite-ui';

export const HeightContainer = styled.div`
  height: 100%;
`;

export const Fixed = styled.div`
  position: fixed;
  bottom: 56px;
  width: 100%;
`;

export const ToastContainer = styled(MobileContentWrapper)`
  padding: 16px;

  > * {
    margin-top: 8px;
  }
`;
