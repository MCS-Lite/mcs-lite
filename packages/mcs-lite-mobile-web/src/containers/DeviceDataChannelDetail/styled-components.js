import styled from 'styled-components';
import { MobileContentWrapper, Small, A } from 'mcs-lite-ui';

export const CardContainer = styled(MobileContentWrapper)`
  padding: 8px;
`;

export const StyledSamll = styled(Small)`
  display: block;
  margin-top: 4px;
  color: ${props => props.theme.color.grayBase};
`;

export const HistoryContainer = styled(MobileContentWrapper)`
  padding: 16px;
`;

export const HistoryHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-bottom: 8px;
`;

export const ResetWrapper = styled(A)`
  display: flex;
  font-size: 24px;
  align-items: center;

  > *:first-child {
    margin-right: 4px;
  }
`;
