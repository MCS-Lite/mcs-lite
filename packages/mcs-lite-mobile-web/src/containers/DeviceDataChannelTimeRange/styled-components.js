import styled from 'styled-components';
import MobileContentWrapper from 'mcs-lite-ui/lib/MobileContentWrapper';

export const Container = styled(MobileContentWrapper)`
  padding: 16px 16px 56px 16px;
`;

export const InputWrapper = styled.div`
  margin-bottom: 8px;

  > *:first-child {
    margin-bottom: 4px;
  }
`;

export const FakeInput = styled.div`
  display: flex;
  align-items: center;
  box-sizing: border-box;
  border-width: 1px;
  border-style: solid;
  border-radius: 3px;
  padding: 0 10px;
  cursor: pointer;
  min-height: ${props => props.theme.height.normal};
  color: ${props => props.theme.color.black};
  background-color: ${props => props.theme.color.white};
  border-color: ${props => props.theme.color.grayDark};
  font-size: ${props => props.theme.fontSize.p};
  flex-shrink: 0;
  flex: 1;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  padding: 8px 16px;

  > * {
    flex: 1;
  }

  > *:first-child {
    margin-right: 8px;
  }
`;
