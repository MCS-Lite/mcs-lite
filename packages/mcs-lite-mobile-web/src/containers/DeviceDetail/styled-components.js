import styled from 'styled-components';
import { Img, MobileContentWrapper } from 'mcs-lite-ui';
import { Link } from 'react-router';

export const Container = styled(MobileContentWrapper)`
  padding: 4px;
`;

export const StyledImg = styled(Img)`
  height: 192px;
`;

export const CardWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;

  > * {
    height: initial;
    padding: 8px 16px;
    margin: 4px;
    flex-basis: 100%;
  }

  > [data-width ~= half] {
    flex-grow: 1;
    flex-basis: 40%;
  }
`;


export const StyledLink = styled(Link)`
  display: block;
  text-decoration: none;
  cursor: pointer;
  color: currentColor;
`;
