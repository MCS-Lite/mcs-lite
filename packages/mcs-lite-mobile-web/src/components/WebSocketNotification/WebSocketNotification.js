import React from 'react';
import styled from 'styled-components';
import { MobileContentWrapper, Notification, Button, P } from 'mcs-lite-ui';
import { withGetMessages } from 'react-intl-inject-hoc';
import Transition from 'react-motion-ui-pack';
import messages from './messages';

const Fixed = styled.div`
  position: fixed;
  left: 0;
  right: 0;
  top: ${props => props.theme.mobile.headerHeight};
  margin-top: 16px;
`;

const Wrapper = styled(MobileContentWrapper)`
  padding: 0 16px;
`;

const WebSocketNotification = ({ onClick, getMessages: t }) =>
  <Fixed>
    <Wrapper>
      <Transition
        component={false}
        appear={{ opacity: 0.5, translateY: -40 }}
        enter={{ opacity: 1, translateY: 0 }}
      >
        <Notification key="notification">
          <P>{t('info')}</P>
          <Button onClick={onClick}>{t('reconnect')}</Button>
        </Notification>
      </Transition>
    </Wrapper>
  </Fixed>;

export default withGetMessages(messages, 'WebSocketNotification')(WebSocketNotification);
