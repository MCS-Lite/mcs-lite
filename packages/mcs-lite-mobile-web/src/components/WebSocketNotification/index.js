import { withGetMessages } from 'react-intl-inject-hoc';
import compose from 'recompose/compose';
import messages from './messages';
import WebSocketNotification from './WebSocketNotification';

export default compose(
  withGetMessages(messages, 'WebSocketNotification'),
)(WebSocketNotification);
