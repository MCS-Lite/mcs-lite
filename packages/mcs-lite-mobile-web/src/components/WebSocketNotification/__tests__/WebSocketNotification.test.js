import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import R from 'ramda';
import { ThemeProvider } from 'styled-components';
import mobileTheme from '../../../utils/mobileTheme';
import WebSocketNotification from '../WebSocketNotification';

it('should renders <WebSocketNotification> correctly', () => {
  const wrapper = shallow(
    <WebSocketNotification
      getMessages={R.identity}
      onClick={() => {}}
    />,
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});


it('should handle onClick', () => {
  const mockOnClick = jest.fn();
  const wrapper = mount(
    <ThemeProvider theme={mobileTheme}>
      <WebSocketNotification
        getMessages={R.identity}
        onClick={mockOnClick}
      />
    </ThemeProvider>,
  );

  // Before eventHandler with submit type
  expect(mockOnClick).not.toHaveBeenCalled();

  // After eventHandler with submit type
  wrapper.find('button').simulate('click');
  expect(mockOnClick).toHaveBeenCalled();
});
