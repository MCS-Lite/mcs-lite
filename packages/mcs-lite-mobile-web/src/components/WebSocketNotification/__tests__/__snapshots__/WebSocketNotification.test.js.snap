// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders <WebSocketNotification> correctly 1`] = `
<styled.div>
  <Styled(MobileContentWrapper)>
    <Transition
      appear={
        Object {
          "opacity": 0.5,
          "translateY": -40,
        }
      }
      component={false}
      enter={
        Object {
          "opacity": 1,
          "translateY": 0,
        }
      }
      leave={
        Object {
          "opacity": 0,
        }
      }
      onEnter={[Function]}
      onLeave={[Function]}
      runOnMount={true}
    >
      <Notification>
        <P>
          info
        </P>
        <Button
          active={false}
          block={false}
          component="button"
          kind="primary"
          onClick={[Function]}
          round={false}
          size="normal"
          square={false}
        >
          reconnect
        </Button>
      </Notification>
    </Transition>
  </Styled(MobileContentWrapper)>
</styled.div>
`;
